src/backup.ts

```diff
@@ -0,0 +1,482 @@
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const newExpireTime = new PreciseDate(Date.now() + oneDay);
+   * await backup.updateExpireTime(newExpireTime);
+   */
+  updateExpireTime(
+    expireTime: PreciseDate,
+    gaxOptionsOrCallback?: CallOptions | UpdateExpireTimeCallback,
+    cb?: UpdateExpireTimeCallback
+  ): void | Promise<Backup> {
+    const callback =
+      typeof gaxOptionsOrCallback === 'function'
+        ? (gaxOptionsOrCallback as UpdateExpireTimeCallback)
+        : cb;
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object'
+        ? (gaxOptionsOrCallback as CallOptions)
+        : {};
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IUpdateBackupRequest = {
+      backup: {
+        name: this.formattedName_,
+        expireTime: expireTime.toStruct(),
+      },
+      updateMask: {
+        paths: ['expire_time'],
+      },
+    };
+    return this.request<databaseAdmin.spanner.admin.database.v1.IBackup>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'updateBackup',
+        reqOpts,
+        gaxOpts,
+      },
+      (err, response) => {
+        callback!(err, response);
+      }
+    );
+  }
+
+  delete(): Promise<void>;
+  delete(gaxOptions?: CallOptions): Promise<void>;
+  delete(callback: DeleteCallback): void;
+  delete(gaxOptions: CallOptions, callback: DeleteCallback): void;
+  /**
+   * Deletes a backup.
+   *
+   * @method Backup#delete
+   * @param {object} [gaxOptions] Request configuration options, outlined here:
+   *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+   * @param {DeleteBackupCallback} [callback] Callback function.
+   * @returns {Promise<void>} when resolved, the backup will have been deleted.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * await backup.delete();
+   */
+  delete(
+    gaxOptionsOrCallback?: CallOptions | DeleteCallback,
+    cb?: DeleteCallback
+  ): void | Promise<void> {
+    const callback =
+      typeof gaxOptionsOrCallback === 'function'
+        ? (gaxOptionsOrCallback as DeleteCallback)
+        : cb;
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object'
+        ? (gaxOptionsOrCallback as CallOptions)
+        : {};
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IDeleteBackupRequest = {
+      name: this.formattedName_,
+    };
+    return this.request<databaseAdmin.spanner.admin.database.v1.IBackup>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'deleteBackup',
+        reqOpts,
+        gaxOpts,
+      },
+      err => {
+        callback!(err, null);
```

Comment:
Do we need to pass "null" in the callback, or can we just "callback(err)"?

Reply:
Good catch. It wasn't necessary.